app/static/src/app/apiService.ts

```diff
@@ -139,6 +139,33 @@ export class APIService<S extends string, E extends string> implements API<S, E>
         this._verifyHandlers(url);
         return this._handleAxiosResponse(axios.get(url, { headers: this._headers }));
     }
+
+    async getScript<T>(url: string): Promise<void | T> {
+        this._verifyHandlers(url);
+
+        const reqHeader = "Accept";
+        const respHeader = "content-type"; // Express lower-cases all headers
+        const headerValue = "application/javascript";
+
+        const headers = { ...this._headers };
+        headers[reqHeader] = headerValue;
+        return axios.get(url, { headers }).then((axiosResponse: AxiosResponse) => {
+            if (!axiosResponse.headers[respHeader] || !axiosResponse.headers[respHeader].startsWith(headerValue)) {
+                const errorMsg = `Response from ${url} must have ${respHeader}: ${headerValue} to get as script`;
+                this._commitError(APIService.createError(errorMsg));
+                return null;
+            }
+            const script = axiosResponse.data;
+            const result = eval(script);
```

Comment:
`eval` is bad practice unless you trust where your script came from - which we do. The alternative to `eval` would be to fetch the script by inserting `<script>` tags into the DOM. This felt like a difficult approach to fit into the Vue/Vuex/Typescript model where we want to keep localised control of the responses, and also felt like it would get messy when doing multiple calls to the model endpoint. Also don't think we could get that to work with a POST anyway.